assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py

```diff
@@ -85,17 +98,35 @@ def mdc_preprocessor(
         )
 
     # Output MLTable
-    first_data_row = df.select("data").rdd.map(lambda x: x).first()
+    first_data_row = df.select(MDC_DATA_COLUMN).rdd.map(lambda x: x).first()
 
     spark = init_spark()
-    data_as_df = spark.createDataFrame(pd.read_json(first_data_row["data"]))
+    data_as_df = spark.createDataFrame(pd.read_json(first_data_row[MDC_DATA_COLUMN]))
 
-    def tranform_df_function(iterator):
+    if extract_correlation_id is True:
+        # Add emtpy column to get the correlationId in the schema
+        data_as_df = data_as_df.withColumn(MDC_CORRELATION_ID_COLUMN, lit(""))
+
+    def tranform_df_function_with_correlation_id(iterator):
+        for df in iterator:
+            yield pd.concat(
+                extract_data_and_correlation_id(getattr(row, MDC_DATA_COLUMN), getattr(row, MDC_CORRELATION_ID_COLUMN)) for row in df.itertuples() # noqa
+            )
+
+    def extract_data_and_correlation_id(entry, correlationid):
+        result = pd.read_json(entry)
+        result[MDC_CORRELATION_ID_COLUMN] = ""
+        for index, row in result.iterrows():
+            result.loc[index, MDC_CORRELATION_ID_COLUMN] = correlationid + "_" + str(index)
```

Comment:
oh I see you ended up implementing this right away!

Reply:
Yes this has all the MDC changes